source/val/validate_image.cpp

```diff
@@ -1729,6 +1729,14 @@ spv_result_t ValidateImageQuerySizeLod(ValidationState_t& _,
     return _.diag(SPV_ERROR_INVALID_DATA, inst) << "Image 'MS' must be 0";
   }
 
+  const auto target_env = _.context()->target_env;
+  if (spvIsVulkanEnv(target_env)) {
+    if (info.sampled != 1) {
+      return _.diag(SPV_ERROR_INVALID_DATA, inst)
```

Comment:
Probably worth adding the Vulkan VUID for this error (and using it's language). VUID is VUID-StandaloneSpirv-OpImageQuerySizeLod-04659. Change the error to:
> OpImageQuerySizeLod must only consume an “Image” operand whose type has its “Sampled” operand set to 1